kouncil-frontend/libs/schema-registry/src/lib/avro/avro-utils.service.ts

```diff
@@ -34,13 +49,16 @@ export class AvroUtilsService {
     return example;
   }
 
-  private getRandomValueBasedOnType(type: string): number | string | boolean | null {
+  private getRandomValueBasedOnType(type: string): number | string | boolean | null | Date | ArrayBuffer {
```

Comment:
Ok, this method is growing and growing. How about this solution:
```
  private readonly typeFunctionMap: any = {
    'int': this.getRandomInt,
    'long': this.getRandomInt,
    'double': this.getRandomFloat,
    'float': this.getRandomFloat
    ...
  };
```

Then you can use it in processField.

Reply:
Ofc replace any with the appropriate type.